### PR TITLE
Numba support

### PR DESCRIPTION
**Pull request by mehrdadn**
A simple fix for addressing #593.
(Please let tests run before merging, just to make sure.)

### CODE REVIEW

**Comment by robertnishihara** [comment]
ok to test

**Comment by robertnishihara** [comment]
Shouldn't this already work out of the box (other than the `inspect.isfunction` problem referred to in #593)?

For example, the following works.

```python
```

File: python/ray/worker.py [hunk 1]
```diff
@@ -1914,6 +1914,13 @@ def remote(*args, **kwargs):
 
   def make_remote_decorator(num_return_vals, num_cpus, num_gpus, func_id=None):
     def remote_decorator(func_or_class):
+      try:
+        # Handle the Numba case: if this is a Numba-compiled function, get the underlying Python function
+        func_or_class = getattr(func_or_class, func_or_class.__numba__)
```
Comment (robertnishihara) [inline python/ray/worker.py]: Wouldn't this essentially remove the jitting? So we wouldn't get the benefits of numba? Or am I misinterpreting it?

**Comment by mehrdadn** [comment]
Ooh, yeah, you're right. It didn't occur to me to check if `Dispatcher` might define `__reduce__`, but apparently it does. In that case yeah you wouldn't need this, never mind :) thanks for the catch!

File: python/ray/worker.py (hunk 1 above)
Comment (mehrdadn) [inline python/ray/worker.py]: Nah you're right, I was only trying to fix the error in the other issue. Didn't even cross my mind that we might need to actually jit it afterward haha.

**Pull request closed without merging**